### DISCUSSION TITLE
No SAML in Element X?

### DISCUSSION THREAD

**Discussion by teward** [element-hq/element-meta#2616] [Ideas]
Element X on mobile devices, currently advertised heavily by Element, has no SAML / Enterprise SSO login support.  This is a MAJOR feature deficit compared to standard Element client functionality.

Is this intentionally missing functionality or just 'not developed'?

**Comment by rkokkelk** [comment]
Unfortunately no change so far. Given the recently rebranding of the old element app to _element classic_, I've tried to switch to element X. But cannot change until SSO is implemented.

**Comment by Powerkrieger** [comment]
This cant be just a problem between 5 people right? My company is going to switch to matrix and we will all have to use element instead of element x -.-

**Reply by davidmehren** [reply]
Element X supports SSO via next-gen auth with Matrix Authentication Service (MAS), which in turn [supports upstream auth](https://element-hq.github.io/matrix-authentication-service/setup/sso.html).
